### PR TITLE
Fix for poor performance in Angle.__str__ and Angle._repr_latex_

### LINKED ISSUE #6995: Very poor performance with Angle.__str__ and Angle._repr_latex_
The following illustrates very poor performance with ``Angle.__str__`` and ``Angle._repr_latex_`` with arrays:

```python
In [15]: a = Angle(np.ones(1000000), u.deg)

In [16]: %time a.__repr__()
CPU times: user 324 µs, sys: 23 µs, total: 347 µs
Wall time: 332 µs
Out[16]: '<Angle [ 1., 1., 1.,...,  1., 1., 1.] deg>'

In [17]: %time a.__str__()
CPU times: user 22.4 s, sys: 341 ms, total: 22.8 s
Wall time: 23.4 s
Out[17]: "['1d00m00s' '1d00m00s' '1d00m00s' ..., '1d00m00s' '1d00m00s' '1d00m00s']"

In [18]: %time a._repr_latex_()
CPU times: user 22.6 s, sys: 436 ms, total: 23.1 s
Wall time: 23.7 s
Out[18]: '[$1^\\circ00{}^\\prime00{}^{\\prime\\prime}$\n $1^\\circ00{}^\\prime00{}^{\\prime\\prime}$\n $1^\\circ00{}^\\prime00{}^{\\prime\\prime}$ ...,\n $1^\\circ00{}^\\prime00{}^{\\prime\\prime}$\n $1^\\circ00{}^\\prime00{}^{\\prime\\prime}$\n $1^\\circ00{}^\\prime00{}^{\\prime\\prime}$]'
```

This poor performance is unnecessary since only a few items are returned in the strings, and the rest is ``...``

This is not an issue with Quantity or Distance, but does percolate up to e.g. SkyCoord I think.

### PR DESCRIPTION
**Pull request by vermashresth**
Changed the call to `to_string()` method inside `__str__()`  and `_repr_latex_()` to be done through numpy machinery to prevent unnecessary computing and poor performance.

Edit: close #6995

### CODE REVIEW

**Comment by bsipocz** [comment]
@astrofrog - do we need changelog for this? I would guess not, but it's your call.

**Comment by astrofrog** [comment]
I think we should add a changelog entry for 2.0.4 - I also think it would make sense to make sure we have a test for this to make sure this works correctly (can't check this right now)

**Comment by mhvk** [comment]
Don't know that I see the point in a change-log entry: all this does is speed up what was there. There is also a test of sorts already: https://github.com/astropy/astropy/blob/master/astropy/coordinates/tests/test_angles.py#L867

**Comment by vermashresth** [comment]
@eteq Sure, I can do this soon. Would the test cases for `__str__` be similar to the ones in `test_repr_latex()`? I think that would cover the else case as well as summarizing for large array cases.

**Comment by mhvk** [comment]
@vermashresth - yes, the test should be (very) similar.

File: astropy/coordinates/tests/test_angles.py [hunk 1]
```diff
@@ -846,6 +846,25 @@ def test_wrap_at_without_new():
...
+    # scalar angle
+    scangle = Angle('10.2345d') 
+    strscangle = scangle.__str__()
+
+    # non-scalar array angles
+    arrangle = Angle(['10.2345d', '-20d'])
+    strarrangle = arrangle.__str__()
+
+    assert strarrangle == '[10d14m04.2s -20d00m00s]'
+    assert strscangle in strarrangle
```
Comment (mhvk) [inline astropy/coordinates/tests/test_angles.py]: Looks good, but can you just test `stsscangle` directly? Maybe put it just below were it is defined.

Comment (vermashresth) [reply astropy/coordinates/tests/test_angles.py]: @mhvk Okay I will do, any other cases that should be checked?

**Comment by mhvk** [comment]
@vermashresth - this looks good to me. I made another commit to remove spaces (that github editor is darn annoying!). Now can be merged if tests pass. Thanks!

**Comment by bsipocz** [comment]
Tests passed, so I'm merging (the failing doc build is unrelated and is being dealt with separately).

**Comment by eteq** [comment]
Of course this already got merged, but it looked good, so I'm happy.  Thanks @vermashresth !

**Pull request merged**